OutSystems/outsystems-maps
# RGRIDT-888: [Maps] Drawing Tools - Draw Marker Block

This PR is for RGRIDT-888: [Maps] Drawing Tools - Draw Marker Block

### What was happening
* As an OS developer, I want to enable my end-users to put markers on a Map in my App, so that they can highlight relevant information on that Map.

### What was done
* Added a new DrawMarker block responsible for creating the new markers on the map using the drawing tools.
* Added a new Config for the draw markers that extends the DrawConfig (general drawing tools configurations)
* Added the subscriber for the tool events as well as the unsubscriber
* Added the new drawing tool to the Factory
* Added the new method **AddTool** that is available as an API as well as the RemoveTool
* Added the ChangeToolProperty and the GetDrawingToolsByToolUniqueId which finds the closest drawingTool from a tool
* Created a set of new abstract classes / classes:
  * AbstractTool - responsible for all the OSFramwork tools
  * AbstractProviderTool - responsible for all the GoogleProvider tools
  * DrawMarker - responsible for all the GoogleProvider drawing markers tools
  
![image](https://user-images.githubusercontent.com/6432232/134025992-16e12fd0-d827-478e-bc97-c2bb363112fe.png)

* Added a new GenerateUniqueId method that allows generating new uniqueIds for the new shape elements that are going to be created by the drawingTools. We need this uniqueId to allow the developer to later use APIs over the element (shape/marker) that is the owner of that uniqueId.

### Test Steps
Test Case | Test steps
-- | --
Platform “if” false DrawMarker | 1. Select the "Polyline" Button<br>2. Toggle Off the "Show/Hide Tool"<br>**Expected: The drawPolyline block should not appear on the DrawingTools**
Platform “if” true DrawMarker | 1. Select the "Polyline" Button<br>2.Toggle Off the "Show/Hide Tool"<br>3.Toggle On the "Show/Hide Tool"<br>**Expected: The drawPolyline block should appear on the DrawingTools**
DrawMarker AllowDrag | 1. Toggle On the AllowDrag<br>2. Select the DrawMarker tool on the drawing tools (hover the map)<br>3. Draw a Marker on the Map<br>**Expected: A new marker should appear on the Map and you should be able to drag and drop it**
DrawMarker AllowDrag False | 1. Toggle Off the AllowDrag<br>2. Select the DrawMarker tool on the drawing tools (hover the map)<br>3. Draw a Marker on the Map<br>**Expected: A new marker should appear on the Map and you should not be able to drag and drop it**
DrawMarker IconUrl | 1. Change the iconUrl to "Info"<br>2. Select the Marker tool on the drawing tools (hover the map)<br>3. Draw a Marker on the Map<br>**Expected: A new marker should appear on the Map with the icon "i"**

### Screenshots
![markerDraw](https://user-images.githubusercontent.com/6432232/134028411-7d788b1f-ee3a-418b-af7d-dbab5e38582b.gif)

### Checklist
* [x] requires changes in OutSystems (if so, provide a module with changes) - RGRIDT-882
